unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/FileSeriesImporter.cs

```diff
@@ -1,91 +1,147 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using UnityEditor;
 using UnityEngine;
 
-//A class for importing a STL series from a dir
+//A class for importing a STL series from a directory.
 public class FileSeriesImporter
 {
-    private GameObject seriesGameObject = new GameObject();
-    private Mesh mesh = new Mesh();
+    public GameObject ModelGameObject { get; private set; } = new GameObject();
+    public Mesh Mesh { get; private set; } = new Mesh();
 
     private string[] filePaths;
     private string extension;
 
-    private bool cancelConvertion = false;
-    private bool equalTopology = false; 
-
-    public GameObject GetGameObject()
-    {
-        return seriesGameObject;
-    }
-
-    public Mesh GetMesh()
-    {
-        return mesh;
-    }
+    private FileImporter fileImporter;
+    private Dictionary<string, Vector3> boundingVertices = new Dictionary<string, Vector3>();
     
     // Object constructor, initiates file series import.
     public FileSeriesImporter(string rootDirectory)
     {
         GetFiles(rootDirectory);
         GetGameObjectName(rootDirectory);
         LoadFiles();
+        ConfigureMesh();
     }
 
+    //Loads file paths of particular frames and their extension
     private void GetFiles(string rootDirectory)
     {
         filePaths = Directory.GetFiles(rootDirectory + @"\");
         extension = Path.GetExtension(filePaths[0]);

```

Comment:
We should use first known extentions not first from all files.
Please log also information if there are more than 1 extension in the directory.

Reply:
I'm assuming that in the folder are only files with the needed meshes. I've added consistency check, that ensures that every file is the same format as the first one. I'll add it to import requirements along with VTK format specification.